app/lib/frontend/handlers.dart

```diff
@@ -29,21 +30,36 @@ void _logPubHeaders(shelf.Request request) {
 /// The passed in [shelfPubApi] handler will be used for handling requests to
 ///   - /api/*
 shelf.Handler createAppHandler(shelf.Handler shelfPubApi) {
+  final legacyDartdocHandler = LegacyDartdocService().router.handler;
   final pubDartlangOrgHandler = PubDartlangOrgService().router.handler;
   final pubSiteHandler = PubSiteService(shelfPubApi).router.handler;
   return (shelf.Request request) async {
     _logPubHeaders(request);
 
+    // legacy dartdocs.org URLs
+    final host = request.requestedUri.host;
+    if (host == 'www.dartdocs.org' || host == 'dartdocs.org') {
+      final rs = await legacyDartdocHandler(request);
+      return rs ?? shelf.Response.notFound('Not Found.');
```

Comment:
Let's move `shelf.Response.notFound('Not Found.')` into `LegacyDartdocService`.

Using a catch-all rule at the end of the class..